witte_jacquelyn_ndvi.py

```diff
@@ -0,0 +1,898 @@
+# * Keep any output files SEPARATE FROM input files. Outputs should be created in an outputs directory that is created in the code (if needed) and/or tested for.
+# * Use the functions that we demonstrated during class to make your workflow more efficient.
+# * BONUS - if you  chose - you can export your data as a csv file. You will get bonus points for doing this.
+# 
+# 
+# ## Assignment Requirements
+# 
+# Your submission to the GitHub repository should include:
+# * This Jupyter Notebook file (.ipynb) with:
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
+# # Pseudocode for this workflow
+- There are three required files to get a single mean NDVI value:
+    1. Landsat tif
+    2. Landsat QC tif
+    3. Landsat shapefile
+
+The basic flow to calculate the mean NDVI for a single date of measurements:
+- Read the image tifs
+- Read the QA tif
+- Read the shapefile
+- Merge the individual tif bands into a single xarray
+- Crop the dataArray using the shapefile
+- Clean the cropped dataArray - remove 0's and high values
+- Mask the dataArray using the QA tif
+- Calculate the NDVI
+- Calculate the mean NDVI
+- Turn it into a dataFrame
+- Plot it
+
+The idea is to create many functions to perform most of the outlined steps above. When looping through all the data directories to calculate a single mean NDVI value and grab it's metadata, the sequence of code will be a series of function calls only.
+
+Note, that I am choosing to read all the bands so there is the option to look at RGB or CIR in the future. 
+
+Function 1 - Retrieve tif files
+
+    bands_tif = sorted(os.path.join(landsat_path,'*band[2-5]*.tif'))
+    qa_tif = os.path.join(landsat_path,'*qa*.tif')
+
+Function 2 - Key metadata: date and site
+
+    site = filepath.split('/')[2]
```

Comment:
Would using the "/" here cause issues with other OS'? Is there a way to accomplish this without using this operator?

Reply:
I figured out how to ignore the slashes! 
Reference: https://stackoverflow.com/questions/22804002/how-to-split-path-with-slashes